### ISSUE TITLE
why do I need to download yolov8n.pt， When I train the segmentation model, all parameters have been specified, 

### ISSUE THREAD

**Issue by minwim** [ultralytics/ultralytics#4107]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

when 

Ultralytics YOLOv8.0.145 🚀 Python-3.8.17 torch-2.0.1+cu117 CUDA:0 (NVIDIA RTX A4000, 16117MiB)
[34m[1mengine/trainer: [0mtask=segment, mode=train, model=/home/govion/data/wlm/workspace/models/best.pt, data=/home/disk2/data/data/admin/fragmentCollect/10/yoloV8/data_138.yaml, epochs=2, patience=150, batch=4, imgsz=1240, save=True, save_period=-1, cache=False, device=[0], workers=4, project=/home/disk2/data/data/admin/workDir/138/20230802150301, name=work_dir, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=/home/disk2/data/data/admin/workDir/138/20230802150301/work_dir
Overriding model.yaml nc=4 with nc=5

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]              
  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    591360  ultralytics.nn.modules.block.C2f             [768, 256, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 16                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 19                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1   1969152  ultralytics.nn.modules.block.C2f             [768, 512, 1]                 
 22        [15, 18, 21]  1   2772479  ultralytics.nn.modules.head.Segment          [5, 32, 128, [128, 256, 512]] 
YOLOv8s-seg summary: 261 layers, 11792031 parameters, 11792015 gradients

Transferred 411/417 items from pretrained weights
[34m[1mAMP: [0mrunning Automatic Mixed Precision (AMP) checks with YOLOv8n...
_Downloading https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt to 'yolov8n.pt'..._

**Comment by xaristeidou** [comment]
@minwim By default when you train any model using Ultralytics package, it downloads the `yolov8n` model to run a check for Automatic Mixed Precision (AMP) to ensure the AMP runs correctly, nothing to worry about. For the training procedure the model that will be used is the model you have specified, e.g., `model = YOLO("yolov8x-seg.pt")`.

**Comment by glenn-jocher** [comment]
Hello @minwim,

Precisely, the downloading of 'yolov8n.pt' you are seeing is part of a routine check made by the Ultralytics package to ensure proper functionality of Automatic Mixed Precision (AMP). This check is performed regardless of the specific model you are using or training, hence you glimpsed yolov8n.pt being downloaded.

However, rest assured that your segmentation model training would proceed with the use of your specified model, as indicated in the model initialization line. The 'yolov8n.pt' file isn't directly applied in the training process of your segmentation model unless specified as such.

Please feel free to reach out if you have further questions. 

Best regards,
Glenn Jocher

**Comment by minwim** [comment]
thanks  @xaristeidou @glenn-jocher

**Comment by xbkaishui** [comment]
> Hello @minwim,
> 
> Precisely, the downloading of 'yolov8n.pt' you are seeing is part of a routine check made by the Ultralytics package to ensure proper functionality of Automatic Mixed Precision (AMP). This check is performed regardless of the specific model you are using or training, hence you glimpsed yolov8n.pt being downloaded.
> 
> However, rest assured that your segmentation model training would proceed with the use of your specified model, as indicated in the model initialization line. The 'yolov8n.pt' file isn't directly applied in the training process of your segmentation model unless specified as such.
> 
> Please feel free to reach out if you have further questions.
> 
> Best regards, Glenn Jocher

Hi, can we make this check logic optional, I saw it will download other resource for example font

**Comment by glenn-jocher** [comment]
Hello @xbkaishui,

Thank you for your suggestion. As of now, the AMP check and downloading of resources like fonts is an integral part of the Ultralytics package setup process to ensure consistent performance and visualization across different machines. It guarantees the toolset is completely prepared for whatever task is at hand.

However, your suggestion to make this check logic optional is understandable and appreciated. It's always important for us to consider the user's different contexts and needs. We'll take note of this feedback for future enhancements of the software package.

Feel free to continue sharing your ideas or any issues you may encounter. Your input is quite valuable.

Best Regards,
Glenn Jocher

**Comment by ei-leenx** [comment]
# from ultralytics import YOLO

# Load a model
model = YOLO("yolov8n.pt")

# Train the model
train_results = model.train(
    data="coco8.yaml",  # path to dataset YAML
    epochs=3,  # number of training epochs
    imgsz=587,  # training image size
    device="cpu",  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
)

# Evaluate model performance on the validation set
metrics = model.val()

# Perform object detection on an image
results = model('/content/drive/My Drive/images/train')
results[0].show()

# Export the model to ONNX format
path = model.export(format="onnx")return path to exported model                                                                                                                                               someone help me identify the error in this code

**Comment by glenn-jocher** [comment]
Your code appears mostly correct, but there is a small issue in the final line. The `return` statement is unnecessary and causes a syntax error. Simply remove `return path to exported model` and your script should work as intended. If the issue persists, please share any specific error messages you're encountering for further assistance.

**Comment by ei-leenx** [comment]
> Your code appears mostly correct, but there is a small issue in the final line. The `return` statement is unnecessary and causes a syntax error. Simply remove `return path to exported model` and your script should work as intended. If the issue persists, please share any specific error messages you're encountering for further assistance.

thanks alot, you are a life saver

**Comment by ei-leenx** [comment]
how do I annotate multiple pictures using the free
 CVAT  under one task

**Comment by glenn-jocher** [comment]
You can annotate multiple images in CVAT under one task by creating a new task and uploading all the images you want to annotate into it. CVAT will group these images under the same task, allowing you to annotate them collectively. For more details, consult the [CVAT documentation](https://opencv.github.io/cvat/).

**Comment by ei-leenx** [comment]
https://colab.research.google.com/drive/1InfMxbgASsjc3M2YXSGs8TvTSVDkzjLl#scrollTo=oiYG6I3gmkXL&line=1&uniqifier=1, how to run code on a webcam and test the realtime video in CVAT

**Comment by glenn-jocher** [comment]
To test real-time video using a webcam with CVAT, you'll need to stream video frames to CVAT for annotation. CVAT itself is designed for annotation, not real-time inference. For running real-time detections, consider using the Ultralytics YOLO models directly in Python or via CLI. Refer to [Ultralytics YOLO Docs](https://docs.ultralytics.com/) for guidance on webcam integration.
